pkg/controller/cloud/node_controller.go

```diff
@@ -184,15 +184,6 @@ func (cnc *CloudNodeController) updateNodeAddress(node *v1.Node, instances cloud
 			}
 		}
 	}
-	// If nodeIP was suggested by user, ensure that
-	// it can be found in the cloud as well (consistent with the behaviour in kubelet)
-	if nodeIP, ok := ensureNodeProvidedIPExists(node, nodeAddresses); ok {
-		if nodeIP == nil {
-			glog.Errorf("Specified Node IP not found in cloudprovider")
-			return
-		}
-		nodeAddresses = []v1.NodeAddress{*nodeIP}
```

Comment:
sorry I wasn't clear earlier. I think we should still validate that nodeIP suggested by user exists, but we don't need to override the entire slice here with
```
nodeAddresses = []v1.NodeAddress{*nodeIP}
```

Reply:
You can probably just port the changes from https://github.com/kubernetes/kubernetes/pull/63101

Reply:
got it